snuba/clickhouse/formatter.py

```diff
@@ -78,6 +79,11 @@ def visitColumn(self, exp: Column) -> str:
         ret.append(escape_identifier(exp.column_name) or "")
         return self.__alias("".join(ret), exp.alias)
 
+    def visitNestedColumn(self, exp: NestedColumn) -> str:
+        raise ValueError(
+            f"Clickhouse does not support a dictionary style accessor for nested columns: {exp.column_name}"
+        )
```

Comment:
NestedColumn is allowed by Snuba language, but it is not understood by clickhouse so it must be processed before we get to formatting the query. I don't think we should try to figure out a way to format this here since the query would fail anyway.